bokeh/charts/_builder.py

```diff
@@ -76,8 +81,26 @@ class Builder(HasProps):
 
     x_range = Instance(Range)
     y_range = Instance(Range)
+    y_names = Seq(String)
+    x_names = Seq(String)
```

Comment:
I will suggest one other design, in case you prefer it: Creating isolated "data argument handler" classes for each builder subtype, and then letting `Builder` delegate to this to to validation, etc, of input args related to data sources. So,
```
class TimeSeriesBuilder(Builder):
    # HAS a TabularArgsHandler, that Builder uses

class ScatterBuilder(Builder):
    # HAS a TabularArgsHandler, that Builder uses

class Image(Builder):
    # HAS an ArrayArgsHandler, that Builder uses
```
I think either of these approaches is defensible. The main point is just that the kind of data source input and the various args that are needed for each data source type is something that varies between chart types, and we need to express that variance in the design.